### ISSUE TITLE
debug.log CreateFile: The system cannot find the file specified

### ISSUE THREAD

**Issue by Tsenzuk** [microsoft/vscode-eslint#1154]
Issue Type: <b>Bug</b>

When several VSCode Insiders is opened
And the projects are with React.js
Then Creates debug.log in each folder where .jsx file is opened.
content of the debug.log:
[0105/011038.242:ERROR:registration_protocol_win.cc(102)] CreateFile: The system cannot find the file specified. (0x2)
[0105/011059.935:ERROR:registration_protocol_win.cc(102)] CreateFile: The system cannot find the file specified. (0x2)
[0105/011126.411:ERROR:registration_protocol_win.cc(102)] CreateFile: The system cannot find the file specified. (0x2)
[0105/011151.756:ERROR:registration_protocol_win.cc(102)] CreateFile: The system cannot find the file specified. (0x2)
[0105/011239.872:ERROR:registration_protocol_win.cc(102)] CreateFile: The system cannot find the file specified. (0x2)
[0105/011302.338:ERROR:registration_protocol_win.cc(102)] CreateFile: The system cannot find the file specified. (0x2)
[0105/011322.337:ERROR:registration_protocol_win.cc(102)] CreateFile: The system cannot find the file specified. (0x2)
[0105/011401.478:ERROR:registration_protocol_win.cc(102)] CreateFile: The system cannot find the file specified. (0x2)
[0105/011440.340:ERROR:registration_protocol_win.cc(102)] CreateFile: The system cannot find the file specified. (0x2)

Extension version: 2.1.14
VS Code version: Code - Insiders 1.53.0-insider (98106c48a07d4d07f0f71b4db9b3ff156f223339, 2021-01-01T05:22:15.141Z)
OS version: Windows_NT x64 10.0.19041

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-8550U CPU @ 1.80GHz (8 x 1992)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>protected_video_decode: enabled<br>rasterization: enabled<br>skia_renderer: enabled_on<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.89GB (2.01GB free)|
|Process Argv|--crash-reporter-id f21b2d82-b794-4f79-ba69-28b549a30887|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv695:30137379
vsins829:30139715
vsliv368:30146709
vsreu685:30147344
openlogontheside:30221882
python763:30178808
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
python504:30227505
vswsl492cf:30208930
wsl2prompt:30219162
vstry914:30230485
pythonvsdeb440:30224570
unusedprompt:30219164
folderexplorer:30219166
openfilemenucf:30219169
pythonvsded773:30223139
core-portspanel:30233467

```

</details>

**Comment by dbaeumer** [comment]
@Tsenzuk can you please provide me with a GitHub repository I can clone that demos what you are experiencing.

**Comment by robross0606** [comment]
I am also seeing this issue.  ESlint from cli reports multiple errors in a file, but vscode doesn't show any of them.

**Comment by dbaeumer** [comment]
@robross0606 any steps you can share best via a GitHub repository I can clone

**Comment by Aron4u** [comment]
I have this also, And i figured out it is this repo with a bisect. What I have right now is the svelte next setup
`
npm init svelte@next
`
I did an `npm install`, and a `npm run dev -- --open`

 I just click in the vscode explorer on any file / folder, and at some point it appears. The debug log can appear in the root folder, or any other.

In general I suspect that this is nothing that you can reproduce with a specific repo. My guess would be that this specific to our PCs. But you can try and tell me if it happens for you.

**Comment by dbaeumer** [comment]
@Aron4u thanks for the investigation.

**Comment by Aron4u** [comment]
Damn, I somehow typed something that I did not mean. I meant, that I figured out that its this **extension**, by doing a bisect. 
Sorry for the confusion. 
@dbaeumer Are you able to reproduce this with the svelte next repo? 
otherwise, I could try to get some extra logging for you, if you tell me what you need and how to get it :)

**Comment by dbaeumer** [comment]
From my experience something like this depends on a lot of things including eslint and plugin versions. That is why I usually ask for a repro to ensure that we are looking at the same setup.

**Comment by allardy** [comment]
@dbaeumer We are in the process of switching from TSLint to ESLint in a relatively big codebase, and I have the same issue, but only in one specific package. When I run the CLI, it displays correctly the different warnings and errors, but the ESLint plugin doesn't show anything, and autoFix on save isn't working either.

Usually I run the CLI from the root of the project, which has all the plugins required. The first time I tried to set up ESLint, I added an `.eslintrc.js` in the folder of the project, but I got the following error:
![image](https://user-images.githubusercontent.com/42552874/107471926-06217d80-6b3c-11eb-8646-3ec95af1d28a.png)

At that time, to fix the issue, I used that flag: `yarn run eslint --resolve-plugins-relative-to .`, that fixed my issue, but not the visual cues. I tried to remove the package `react-scripts` (which also removed the plugin) and then everything started working fine.

So yeah, looks like a version mismatch, my version of react-scripts is pretty old (2.1.8) and my guess is that VSCode tries to use the plugin from the current active directory, and not from the location of my `eslint.js` file (which I would expect)

**Comment by dbaeumer** [comment]
@allardy a GitHub repository I can clone with steps to reproduce is always welcome. Please keep in mind that ESLint is very sensitive to the current working directory when resolving things. So the `eslint.workingDirectories` setting might be your friend here.

**Comment by allardy** [comment]
@dbaeumer Clone this branch: https://github.com/botpress/botpress/tree/ya-mig-eslint 
Then remove the config `eslint.nodePath` from `.vscode/settings.json`, then reload the window.

If you open the file `src/bp/ui-admin/src/api.tsx` and change the order of the imports (or type any code that should be fixed by eslint), nothing will be highlighted and if you save, it doesn't fix it. But if you add the config `eslint.nodePath` again, it will work correctly.

In the same folder `src/bp`, you can also try it in `ui-studio`, this package is not affected at all by that change. 

Since we have only a single .eslintrcjs, shouldn't we expect that any invocation will use the libraries in the node_modules closest to the rc file ? I think the worst thing is that the error is completely silent and doesn't give a clue of what's happening, like if there's a version mismatch somewhere

**Comment by SamuelColacchia** [comment]
I was about to post the below then as i was getting my version of vscode insiders I noticed there was a update avaible, on the latest version i am unable to reproduce the issue. 

Version: 1.54.0-insider (system setup)
Commit: 40b3204d70a9c72305c613164272175d0b344b5f
Date: 2021-02-10T17:39:27.208Z
Electron: 11.2.3
Chrome: 87.0.4280.141
Node.js: 12.18.3
V8: 8.7.220.31-electron.0
OS: Windows_NT x64 10.0.19042

---------

I have been encountering the same issue that Tsenzuk mentioned. 

So far I found that opening up a vscode project that has been saved as a workspace or from the recents list causes the issue.

Quick steps to reproduce, im using nestjs to quickly demo it

1. nest n p1 (answer prompts)
2. nest n p2 (answer prompts)
3. In a new window of vscode open p1
4. In a new window of vscode open p2
5. observe that opening any ts files works as one would expect
6. close all vscode windows
7. from "recent folders & workspaces" open p1, opening any ts files works as expected
8. from "recent folders & workspaces" open p2 while p1 remains open in another window, opening a ts file creates a debug.log 

I also created 2 repos to demo this.
[p1](https://github.com/SamuelColacchia/eslintvscodep1)
[p2](https://github.com/SamuelColacchia/eslintvscodep2)
